backend/ibutsu_server/controllers/portal_controller.py

```diff
@@ -0,0 +1,129 @@
+from http import HTTPStatus
+
+import connexion
+
+from ibutsu_server.constants import RESPONSE_JSON_REQ
+from ibutsu_server.db.base import session
+from ibutsu_server.db.models import Portal, User
+from ibutsu_server.filters import convert_filter
+from ibutsu_server.util.query import get_offset
+from ibutsu_server.util.uuid import convert_objectid_to_uuid, is_uuid, validate_uuid
+
+
+def add_portal(portal=None, token_info=None, user=None) -> dict:
+    """Create a portal
+
+    :param body: Portal
+    :type body: dict | bytes
+
+    :rtype: Portal
+    """
+    if not connexion.request.is_json:
+        return RESPONSE_JSON_REQ
+    portal = Portal.from_dict(**connexion.request.get_json())
+    # check if portal already exists
+    if portal.id and Portal.query.get(portal.id):
+        return f"Portal id {portal.id} already exists.", HTTPStatus.CONFLICT
+    user = User.query.get(user)
+    if user:
+        portal.owner = user
+    session.add(portal)
+    session.commit()
+    return portal.to_dict(), HTTPStatus.CREATED
+
+
+@validate_uuid
+def get_portal(id_, token_info=None, user=None) -> dict:
+    """Get a single portal by ID
+
+    :param id: ID of test portal
+    :type id: str
+
+    :rtype: Portal
+    """
+    if not is_uuid(id_):
```

Comment:
Will double check, at this point I was repeating a lot of the patterns from the project controller because the entities should have the same CRUD behaviors and permissions.